### PR TITLE
Fix: remove lines with special code when convert to reduc

### PR DESCRIPTION
**Pull request by hregis**
@eldy 
i push another PR for branch 6, the file has been renamed

### CODE REVIEW

File: htdocs/compta/facture.php [hunk 1]
```diff
@@ -611,7 +611,7 @@
 			$i = 0;
 			foreach ($object->lines as $line)
 			{
-				if ($line->total_ht!=0)
+				if (empty($line->special_code) && $line->total_ht != 0) // Remove lines with special code
```
Comment (eldy) [inline htdocs/compta/facture.php]: Current known special_code are 
1=frais de port
2=ecotax
3=???
In first 2 cases, we need to include the content into the converted discount. Others are unknown...

Which cas do you have to exclude them ?

Comment (hregis) [reply htdocs/compta/facture.php]: @eldy 
"special_code" is used by external module, the value of "special_code" is the module number.
and for my module I also use the field "product_type" with the value 9 for this type of lines. Maybe it is necessary to use "product_type" instead?

Comment (eldy) [reply htdocs/compta/facture.php]: I think a lot of modules has defined a different rule for the field special_code.
I think (in fact i am sure) that some modules does not use a module number here. I found another use cas: 3 is used for options lines by core code itfself.
We can defined a rule like special_code < 100 is used for discount convertion, > 100 not.

product_type is also used by different modules like the subtotal that use value 9 too.
So we can add
"if product_type == 9" to be excluded from discount content

**Comment by hregis** [comment]
@eldy 
I replace with "product_type" and I exclude the lines from the value 9, if you need more I will modify my module.

**Pull request merged**